optuna/samplers/random.py

```diff
@@ -76,6 +76,16 @@ def sample_independent(self, study, trial, param_name, param_distribution):
             v = numpy.round(s / q) * q + param_distribution.low
             # v may slightly exceed range due to round-off errors.
             return float(min(max(v, param_distribution.low), param_distribution.high))
+        elif isinstance(param_distribution, distributions.IntLogUniformDistribution):
+            log_low = numpy.log(param_distribution.low - 0.5)
+            log_high = numpy.log(param_distribution.high + 0.5)
+            s = numpy.exp(self._rng.uniform(log_low, log_high))
+            v = (
+                numpy.round((s - param_distribution.low) / param_distribution.step)
+                * param_distribution.step
+                + param_distribution.low
+            )
+            return int(min(max(v, param_distribution.low), param_distribution.high))
```

Comment:
Unfortunately, we cannot change the order. `IntLogUniformDistribuion` is a subclass of `IntUniformDistribuion` (https://github.com/optuna/optuna/pull/1201#discussion_r421198628), so `isinstance(param_distribution, IntLogUniformDistribution)` should be checked earler than `isinstance(param_distribution, IntUniformDistribution)`. Otherwise, `isinstance(param_distribution, IntLogUniformDistribution)` is never checked.

Reply:
Thank you for your reply. I feel that the `IntLogUniformDistribution` should inherit `BaseDistribution` directly as other distributions do. Doesn't it seem simpler that way? The logic is a bit redundant, but it seems to be easier to maintain.

Reply:
I agree with you. That is easier to maintain for now. `if` blocks could cause an unexpected error by inheriting `IntUniformDistribution`. Even if `IntLogUniformDistribution` inherits `BaseDistribtuion`, less than 10 lines added to the code. That seems acceptable.